### ISSUE TITLE
Implement new notification methodology

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/privacy-pioneer#512]
When I enter "Batman" as a keyword in the Watchlist and then search for "Batman" on Google or Bing, I am not receiving a notification. I know that we had discussed at some point that entering a keyword by the user themselves should not trigger a notification, at least for first parties, because the user knows what they just entered and does not need to be reminded. But when I search for "Batman" on Yelp I actually do get a notification for Yelp collecting "Batman." So, I am not sure. Is this a bug, or is there an explanation?

**Comment by SebastianZimmeck** [comment]
For first party keywords lookups, we should not show notifications. It is clear to the user that they just entered the keyword. @danielgoldelman confirmed that this is our approach.

**Comment by danielgoldelman** [comment]
Conversations on this topic occurred over Teams chat.

If the user had been looking for a SSN appearing in their web traffic instead of the word Batman, it would be important info that the company did not encrypt the data at all when collecting it. Thus, it is a good thing that keywords keep their notifications and are still added as evidence.

**Comment by danielgoldelman** [comment]
As per discussion today, here is the new methodology:

- Notifications for all categories other than general keywords remain the same
- For general keywords:
  - ensure that all keywords generate notifications
  - ensure that all new evidence of each keyword generates a notification
  - no first party keyword should be notified when the first party is a search engine

**Comment by danielgoldelman** [comment]
>>- no first party keyword should be notified when the first party is a search engine

>Either way is fine, but if we make the distinction between search engine sites and non-search engine sites, it should be consistent. i.e., it should not be showing keywords on Google but not showing them on Bing, or sometimes showing them on a search engine and sometimes not, etc.

Ok, then I will not implement this difference.

**Comment by SebastianZimmeck** [comment]
This issue does not seem to be resolved when I test it. There seem to be multiple problems:

### 1. General Keywords
- When I enter "Batman" as a general keyword, then enter "Batman" in Google, I get notified. When I now enter "Spiderman" in the watchlist, then enter "Spiderman" in Google, I do not get notified. I should get notified for new keywords that I enter in the watchlist that subsequently show up in my traffic.
- When I enter "Batman" in Google, I get notified the first time when I search for "Batman." The next time I enter "Batman", I do not get a notification, which is fine. However, I keep hearing the notification sound. That strikes me as odd and is rather annoying because I sometimes even hear it multiple, even when I do a completely new search unrelated to any keyword in the watchlist. Ding ding ding ... every time I type a letter, very annoying.

### 2. IP Address Keywords
- The IP address watchlist item does not seem to consistently work on my end. I enabled the little bell icon in the watchlist, and it also had my correct IP address. Normal browsing did not trigger any notification. When I visited https://whatismyipaddress.com/ I also did not get a notification. I waited 30 seconds. Then, I tried again multiple times and eventually got a notification. I tried refreshing the watchlist page but that did not have any effect. I tried to reproduce getting an IP address, but I am not sure under which circumstances one actually gets a notification and when not.

### 3. Postal Address Keywords
-  I searched on Google for my postal address that was shown in the watchlist but did not get a notification. It sort of went piecemeal: I waited a bit and after 10 seconds I go a notification about my ZIP code being found. Then, I tried on Yahoo, searched for my address, and this time I got region. So, this feature seems rather flaky and partial.

### 4. Email Address Keywords
- Does not seem to work. I searched for my email address, szimmeck@wesleyan.edu, on yahoo.com, waited 30 seconds but did not receive a notification. I refreshed, but still nothing. I clicked on the first link, my Wesleyan page, which also has my email address, but did not receive a notification either.

### 5. Phone Keywords
- Same result as for email does not seem to work.


We need to solve this issue!

@danielgoldelman and @jjeancharles, please work through these points in priority from 1 to 5. This issue is more important than #518 because we still have serious problems.

**Comment by danielgoldelman** [comment]
1. General Keywords
  - I am getting notified, it is just within the other notifications being sent by google
  - The bings are repeated notifications of the same message, which is built in by apple. We keep hearing the notification sound because the keyword continues to be sent when you type a letter (which is how google does autocomplete). This was the main reason why we implemented notifications as we did originally. 
2. I will investigate.
3. The 10 seconds for zip seems to be correct due to our notification wait time methodologies. Yahoo seems to gather region and not postal address. They still show up on the screen, but in HTML format, since it is a search engine with hyperlinks and we are not scanning through main_frame (top level html documents) due to previous decisions to maintain far less false positives
4. investigating
5. ^
